### PR TITLE
[Catalog Promotion] Refactor - rename AllCatalogPromotionsProcessor

### PR DESCRIPTION
**Pull request by ernestWarwas**
| Q               | A
| --------------- | -----
| Branch?         | 1.11 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

AllCatalogPromotionsProcessor renamed to RequestProductVariantCatalogPromotionRecalculate.

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Processor/RequestProductVariantCatalogPromotionRecalculate.php [hunk 1]
```diff
@@ -16,15 +16,15 @@
 use Sylius\Bundle\CoreBundle\CommandDispatcher\ApplyCatalogPromotionsOnVariantsCommandDispatcherInterface;
 use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
 
-final class AllCatalogPromotionsProcessor implements AllCatalogPromotionsProcessorInterface
+final class RequestProductVariantCatalogPromotionRecalculate implements RequestProductVariantCatalogPromotionRecalculateInterface
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Processor/RequestProductVariantCatalogPromotionRecalculate.php:19]: Why did you refactor the name of this service? To be honest, I'm not a fan of the new one 🤔

File: src/Sylius/Bundle/CoreBundle/Listener/CatalogPromotionCreatedListener.php [hunk 2]
```diff
@@ -14,17 +14,17 @@
...
 
 final class CatalogPromotionCreatedListener
 {
     public function __construct(
-        private AllCatalogPromotionsProcessorInterface $allCatalogPromotionsProcessor,
-        private RepositoryInterface $catalogPromotionRepository,
-        private EntityManagerInterface $entityManager
+        private RequestProductVariantCatalogPromotionRecalculateInterface $allCatalogPromotionsProcessor,
+        private RepositoryInterface                                       $catalogPromotionRepository,
+        private EntityManagerInterface                                    $entityManager
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Listener/CatalogPromotionCreatedListener.php:27]: You should revert the arguments aligning here and in other places
```suggestion
        private RepositoryInterface $catalogPromotionRepository,
        private EntityManagerInterface $entityManager
```

File: src/Sylius/Bundle/CoreBundle/Processor/RequestProductVariantCatalogPromotionRecalculate.php (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Bundle/CoreBundle/Processor/RequestProductVariantCatalogPromotionRecalculate.php:19]: To be consistent with other processors from this namespace, maybe sth like:
```suggestion
final class AllProductVariantsCatalogPromotionsProcessor implements AllProductVariantsCatalogPromotionsProcessorInterface
```

File: src/Sylius/Bundle/CoreBundle/spec/Listener/CatalogPromotionEndedListenerSpec.php [hunk 3]
```diff
@@ -35,12 +35,12 @@ function let(
...
-        EntityManagerInterface $entityManager,
-        CatalogPromotionInterface $catalogPromotion,
-        FactoryInterface $stateMachine,
-        StateMachineInterface $stateMachineInterface
+        AllProductVariantsCatalogPromotionsProcessorInterface $catalogPromotionReprocessor,
+        RepositoryInterface                                   $catalogPromotionRepository,
+        EntityManagerInterface                                $entityManager,
+        CatalogPromotionInterface                             $catalogPromotion,
+        FactoryInterface                                      $stateMachine,
+        StateMachineInterface                                 $stateMachineInterface
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/spec/Listener/CatalogPromotionEndedListenerSpec.php:43]: 😡

**Review by AdamKasp — APPROVED** [review]

**Review by Zales0123 — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, @ernestWarwas! :tada:

**Pull request merged**